packages/cli/src/models/files/ZosNetworkFile.ts

```diff
@@ -0,0 +1,418 @@
+    this.data.solidityLibs = solidityLibs;
+  }
+
+  set frozen(frozen: boolean) {
+    this.data.frozen = frozen;
+  }
+
+  set app(app: any) {
+    this.data.app = app;
+  }
+
+  set provider(provider: ProviderInterface) {
+    this.data.provider = provider;
+  }
+
+  set package(_package: PackageInterface) {
+    this.data.package = _package;
+  }
+
+  public setDependency(name: string, { package: thepackage, version, customDeploy }: DependencyInterface = {}) {
+    if (!this.data.dependencies) this.data.dependencies = {};
+
+    const dependency = {
+      package: thepackage,
+      version: semanticVersionToString(version),
+      customDeploy: undefined
+    };
+    if (customDeploy) dependency.customDeploy = customDeploy;
+
+    this.data.dependencies[name] = dependency;
+  }
+
+  public unsetDependency(name: string): void {
+    if (!this.data.dependencies) return;
+    delete this.data.dependencies[name];
+  }
+
+  public updateDependency(name: string, fn: (...args: any[]) => any): void {
+    this.setDependency(name, fn(this.getDependency(name)));
+  }
+
+  // TS-TODO: instance can probably be typed to something interesting.
+  public addContract(alias: string, instance: any, { warnings, types, storage }: { warnings?: any, types?: any, storage?: any } = {}): void {
+    this.setContract(alias, {
+      address: instance.address,
+      constructorCode: constructorCode(instance),
+      bodyBytecodeHash: bytecodeDigest(bodyCode(instance)),
+      localBytecodeHash: bytecodeDigest(instance.constructor.bytecode),
+      deployedBytecodeHash: bytecodeDigest(instance.constructor.binary),
+      types,
+      storage,
+      warnings
+    });
+  }
+
+  public setContract(alias: string, value: ContractInterface): void {
+    this.data.contracts[alias] = value;
+  }
+
+  public unsetContract(alias: string): void {
+    delete this.data.contracts[alias];
+  }
+
+  public setProxies(packageName: string, alias: string, value: ProxyInterface[]): void {
+    const fullname = toContractFullName(packageName, alias);
+    this.data.proxies[fullname] = value;
+  }
+
+  public addProxy(thepackage: string, alias: string, info: ProxyInterface): void {
+    const fullname = toContractFullName(thepackage, alias);
+    if(!this.data.proxies[fullname]) this.data.proxies[fullname] = [];
+    this.data.proxies[fullname].push(info);
+  }
+
+  public removeProxy(thepackage: string, alias: string, address: string): void {
+    const fullname = toContractFullName(thepackage, alias);
+    const index = this._indexOfProxy(fullname, address);
+    if(index < 0) return;
+    this.data.proxies[fullname].splice(index, 1);
+    if(this._proxiesOf(fullname).length === 0) delete this.data.proxies[fullname];
+  }
+
+  public updateProxy(
+    {
+      package: proxyPackageName,
+      contract: proxyContractName,
+      address: proxyAddress
+    }: {
+        package: string,
+        contract: string,
+        address: string
+      },
+    fn: (...args: any[]) => any
```

Comment:
is there a better way to do this to avoid multi-lines here? maybe defining an interface?

Reply:
Definitely. I'm taking note and making this change in a future PR of the port.